### ISSUE TITLE
Support EKS Cluster Upgrades

### ISSUE THREAD

**Issue by brantburnett** [terraform-providers/terraform-provider-aws#6895]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Attempting to change the "version" attribute of an EKS cluster is forcing a new resource. It would be useful for this to instead send an update-cluster-version command to AWS to perform the upgrade.  Downgrades would probably still force a new resource.

### New or Affected Resource(s)

* aws_eks_cluster

### Potential Terraform Configuration

```hcl
resource "aws_eks_cluster" "k8s_primary" {
  name            = "primary"
  role_arn        = "${aws_iam_role.eks.arn}"
  version         = "1.11" # This attribute was missing and defaulted to 1.10 before attempt to apply an upgrade

  vpc_config {
    security_group_ids = ["${aws_security_group.k8s_primary_master.id}"]
    subnet_ids         = ["${aws_subnet.k8s_primary.*.id}"]
  }

  depends_on = [
    "aws_iam_role_policy_attachment.eks-AmazonEKSClusterPolicy",
    "aws_iam_role_policy_attachment.eks-AmazonEKSServicePolicy",
  ]
}
```

### References

* https://aws.amazon.com/about-aws/whats-new/2018/12/amazon-eks-adds-managed-cluster-updates-and-support-for-kubernetes/
* https://docs.aws.amazon.com/eks/latest/userguide/update-cluster.html

**Comment by ewbankkit** [comment]
@brantburnett That was fixed in https://github.com/terraform-providers/terraform-provider-aws/pull/6843 and released in [v1.52.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1520-december-13-2018).

**Comment by brantburnett** [comment]
My apologies, I looked but couldn't find that.  I'm testing it now, I'll close this issue and file a new one if I encounter any bugs.  Thanks.
